iree/compiler/Dialect/HAL/Transforms/RepeatDispatches.cpp

```diff
@@ -0,0 +1,74 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iree/compiler/Dialect/HAL/IR/HALDialect.h"
+#include "iree/compiler/Dialect/HAL/IR/HALOps.h"
+#include "iree/compiler/Dialect/HAL/IR/HALTypes.h"
+#include "mlir/Dialect/SCF/SCF.h"
+#include "mlir/Dialect/StandardOps/IR/Ops.h"
+#include "mlir/Pass/Pass.h"
+
+namespace mlir {
+namespace iree_compiler {
+namespace IREE {
+namespace HAL {
+namespace {
+
+static llvm::cl::opt<int> repeatDispatchNum{
```

Comment:
take this as a pass parameter instead - it'd be nice to reduce the number of global flags we have and not always run this pass regardless of whether the flag is set - test-only things should only show up when they are enabled

as with renaming the pass, if this flag does exist it should be very clear it's for benchmarking (`iree-hal-benchmark-dispatch-repeat-count` or something)

Reply:
I made it to be a pass parameter. I think the flag has to exist. Otherwise I don't have a way to set the value when running `iree-translate`.

I also tried to register it as a pass option. I think all these options will not get linked to `*-translate`.

Reply:
The tradeoff here is compilation time and binary size. It takes more time to compile and the binary size gets much larger.

Reply:
ugh that's annoying - pass options would be so nice if they were usable :P